File: LeetCode Solutions/Binary Tree/Symmetric Tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSymmetric(self, root: TreeNode) -> bool:
        if(not root):
            return True
        def isMirror(p,q) -> bool:
            if(not p and not q):
                return True
            if((p and not q) or (not p and q) or (p.val != q.val)):
                return False
            if(not isMirror(p.left, q.right)):
                return False
            if(not isMirror(p.right, q.left)):
                return False
            return True
        return isMirror(root.left, root.right)

File: LeetCode Solutions/Top Interview Questions(Easy Collection)/Trees/Symmetric Tree/Symmetric Tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public boolean isSymmetric(TreeNode root) {
        if (root == null){
            return true;
        }
        return isSymmetric(root,root);    
    }
    public boolean isSymmetric(TreeNode p, TreeNode q){
        if (p == null && q == null){
            return true;
        }
        if (p != null && q != null && p.val == q.val){
            return isSymmetric(p.left, q.right) && isSymmetric(p.right, q.left);
        }
        return false;
    }
}